ocrd/ocrd/processor/helpers.py

```diff
@@ -69,7 +71,12 @@ def run_processor(
     log.debug("Processor instance %s (%s doing %s)", processor, name, otherrole)
     t0_wall = perf_counter()
     t0_cpu = process_time()
-    processor.process()
+    mem_usage = memory_usage(proc=processor.process, interval=.1, timeout=None, timestamps=True, multiprocess=True, include_children=True)
```

Comment:
```suggestion
    mem_usage = memory_usage(proc=processor.process, 
                            interval=.1, timeout=None, timestamps=True, 
                            # include sub-processes
                            multiprocess=True, include_children=True, 
                            # get proportional set size instead of RSS
                            backend='psutil_pss')
```

Reply:
If it is really that simple – let's make this configurable (PSS vs RSS measuring). Did you try this?

Reply:
Sure, however we first need a mechanism to toggle this behavior. Simple idea would be to add CLI flags `--mem-profile-rss` and `--mem-profile-pss`.

@bertsky prefers a way to have this configurable via the logging configuration - @M3ssman do you have a good idea how that might work? That way, we could keep the CLI arguments small and the `--help` output less confusing.

Reply:
Perhaps for the moment it could be as simple as 
```python
mem_prof = os.getenv('OCRD_PROFILE', '')
if mem_prof:
    ...
        backend='psutil_pss' if mem_prof == 'PSS' else 'psutil')
```

Reply:
I like environment variables in general, but when the choice is between environment variables and CLI flags, I prefer the latter, because at least it's consistent. Because otherwise you could get `OCRD_PROFILE=PSS ocrd-dummy --profile-file my.prof ...`.

Reply:
@bertsky What-how do you mean by configurable via logging-conf?
 
But wait, looks like ocrd might lack a general configuration ini-file, doesn't it? 
I'm using config-file nowadays virtually in every python (client-)application, so thinking in terms of ocrd-as-framework (please don't mess up with the discussion ocrd(-core) as library vs. framework). Even standard python configuration API works pretty well, only have to work around if you want the ConfigParser also evaluate lists and dicts. 
Actually, with the latest python apps I tend to use CLI flags only to specify the path to the config-file :slightly_smiling_face:

Reply:
@bertsky What-how do you mean by configurable via logging-conf?
 
But wait, looks like ocrd might lack a general configuration ini-file, doesn't it? 
I'm using config-file nowadays virtually in every python (client-)application, so thinking in terms of ocrd-as-framework (please don't mess up with the discussion ocrd(-core) as library vs. framework). Even standard python configuration API works pretty well, only have to work around if you want the ConfigParser also evaluate lists and dicts. 
Actually, with the latest python apps I tend to use CLI flags only to specify the path to the config-file :slightly_smiling_face:

Reply:
We tried to avoid a config file for the longest time to eliminate sources of trouble, or rather, force us to keep the interfaces as simple as possible and have sensible defaults instead of making everything configurable.

That being said, we do have `$HOME/.local/share/resource.yml` which is a kind of config file and we have the configuration file for the logging.

But if the CLI become to convoluted with lots of flags and there is consensus, I won't veto a configuration file. Syntax-wise, I'm a big fan of YAML combined with JSON-Schema but ini, TOML, dotenv etc. all are fine with me.

Reply:
@kba I agree, however I did not argue env var _plus_ cli arg, but only the former. Perhaps we could also move `--profile` and `--profile-file` into env var control entirely.

Or we go the other direction and put everything into a single CLI arg, just with multi-faceted value: `--profile mem=rss,cpu,file=foo.log` for example.

For the broader discussion of how to configure processors without adding to many options / parameters and what for, see [this old thread](https://github.com/OCR-D/core/issues/376#issuecomment-562349028)

Reply:
OK, now that the wonky multiple-process-call behavior is fixed, I'll implement this via environment variable `OCRD_PROFILE` and `OCRD_PROFILE_FILE`

* `OCRD_PROFILE` not set or empty: no profiling
* `OCRD_PROFILE_FILE` set to a value (== current `--profile-file` flag):
  * if `OCRD_PROFILE` is defined but empty: no profiling
  * if `OCRD_PROFILE` is not defined: set `OCRD_PROFILE` to `CPU`
  * if `OCRD_PROFILE` is defined but does not contain `CPU`: raise error
* `OCRD_PROFILE` contains `CPU`, do CPU profiling (== current `--profile` flag)
* `OCRD_PROFILE` contains `RSS` do RSS memory profiling
* `OCRD_PROFILE` contains `PSS`: do proportionate mem profiling
* If `OCRD_PROFILE` contains both `PSS` and `RSS`: raise error

Did I catch everything?

Reply:
Implemented in 4fdfbf39b

Reply:
Environment variables look good to me. 

In the readme, I would also suggest to reference an external link where the difference between `RSS` and `PSS` is explained or at least briefly described. May not be the best suggestion, but consider [this](https://elinux.org/Android_Memory_Usage).

Reply:
yes, sounds reasonable and complete AFAICS –  thx!